core/capabilities/vault/gw_handler.go

```diff
@@ -0,0 +1,312 @@
+	metrics, err := newMetrics()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create metrics: %w", err)
+	}
+
+	return &GatewayHandler{
+		capRegistry:    capabilitiesRegistry,
+		secretsService: secretsService,
+		gwConnector:    gwConnector,
+		lggr:           lggr.Named(HandlerName),
+		metrics:        metrics,
+	}, nil
+}
+
+func (h *GatewayHandler) Start(ctx context.Context) error {
+	return nil
+}
+
+func (h *GatewayHandler) Close() error {
+	return nil
+}
+
+func (h *GatewayHandler) ID(ctx context.Context) (string, error) {
+	return HandlerName, nil
+}
+
+func (h *GatewayHandler) HandleGatewayMessage(ctx context.Context, gatewayID string, req *jsonrpc.Request[json.RawMessage]) (err error) {
+	h.lggr.Debugf("Received message from gateway %s: %v", gatewayID, req)
+
+	var response *jsonrpc.Response[json.RawMessage]
+	switch req.Method {
+	case vault_api.MethodSecretsCreate:
+		response = h.handleSecretsCreate(ctx, gatewayID, req)
+	case vault_api.MethodSecretsGet:
+		response = h.handleSecretsGet(ctx, gatewayID, req)
+	default:
+		response = h.errorResponse(ctx, gatewayID, req, api.UnsupportedMethodError, errors.New("unsupported method: "+req.Method))
+	}
+
+	if err = h.gwConnector.SendToGateway(ctx, gatewayID, response); err != nil {
+		h.lggr.Errorf("Failed to send message to gateway %s: %v", gatewayID, err)
+		return err
+	}
+
+	h.lggr.Infof("Sent message to gateway %s: %v", gatewayID, response)
+	h.metrics.requestSuccess.Add(ctx, 1, metric.WithAttributes(
+		attribute.String("gateway_id", gatewayID),
+	))
+	return nil
+}
+
+func (h *GatewayHandler) handleSecretsCreate(ctx context.Context, gatewayID string, req *jsonrpc.Request[json.RawMessage]) *jsonrpc.Response[json.RawMessage] {
+	var requestData vault_api.SecretsCreateRequest
+	if err := json.Unmarshal(*req.Params, &requestData); err != nil {
+		return h.errorResponse(ctx, gatewayID, req, api.UserMessageParseError, err)
+	}
+	h.lggr.Infof("Debugging: handleSecretsCreate 1 %s: %v", gatewayID, req)
+	vaultCapRequest := vault.CreateSecretsRequest{
+		RequestId: req.ID,
+		EncryptedSecrets: []*vault.EncryptedSecret{
+			{
+				Id: &vault.SecretIdentifier{
+					Owner:     requestData.Owner,
+					Namespace: "", // TBD
+					Key:       requestData.ID,
+				},
+				EncryptedValue: requestData.Value,
+			},
+		},
+	}
+	vaultCapResponse, err := h.secretsService.CreateSecrets(ctx, &vaultCapRequest)
+	if err != nil {
+		h.lggr.Infof("Debugging: h.secretsService.CreateSecrets failed, erro: %s", err.Error())
+		return h.errorResponse(ctx, gatewayID, req, api.FatalError, err)
+	}
+	h.lggr.Infof("Debugging: handleSecretsCreate got response. GatewayId: %s, req: %v, Response: %s", gatewayID, req, vaultCapResponse.String())
+
+	vaultResponseProto := &vault.CreateSecretsResponse{}
+	err = protojson.Unmarshal(vaultCapResponse.Payload, vaultResponseProto)
```

Comment:
We should avoid using protojson.Unmarshal because it doesn't marshal data canonically (in fact we should  probably avoid tampering with the payload in any way so we make it easy to reconstruct the signatures)